bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BufferedReadChannel.java

```diff
@@ -86,14 +86,25 @@ public synchronized int read(ByteBuf dest, long pos, int length) throws IOExcept
                 // here we reached eof.
                 break;
             } else {
-                // We don't have it in the buffer, so put necessary data in the buffer
-                readBufferStartPosition = currentPosition;
                 int readBytes = 0;
-                if ((readBytes = validateAndGetFileChannel().read(readBuffer.internalNioBuffer(0, readCapacity),
-                        currentPosition)) <= 0) {
+                try {
+                    // We don't have it in the buffer, so put necessary data in the buffer
+                    readBufferStartPosition = currentPosition;
```

Comment:
We can check the `pos` in advance; if it is negative, throw an exception directly. This pr will invalidate the read buffer cache. Maybe we can reuse the read buffer cache at next time.

Reply:
changed to method early statement to protect negative pos read.